### ISSUE TITLE
Create a simple non-docker version of App

### ISSUE THREAD

**Issue by nssidhu** [Azure-Samples/azure-search-openai-demo-csharp#258]
The current Source code cannot be made to run on local installation without docker.
I would say this has steep learning curve or setup curve. There is lack of explanation as well.
I know the intention of docker was to make it simple, but if docker itself is a problem, this becomes non-starter.
Unless the intention of this project was to impart knowledge to masses and make them understand how the different part of the projects to use, this is failing that achieve objective.

There are millions of developers who are not using docker.

Please provide non-docker version and very good explanation.

======================================================================================

> Please provide us with the following information:
> ---------------------------------------------------------------

### This issue is for a: (mark with an `x`)
```
- [ ] bug report -> please search issues before submitting
- [ ] feature request
- [ ] documentation issue or request
- [ ] regression (a behavior that used to work and stopped in a new release)
```

### Minimal steps to reproduce
>

### Any log messages given by the failure
>

### Expected/desired behavior
>

### OS and Version?
> Windows 7, 8 or 10. Linux (which distribution). macOS (Yosemite? El Capitan? Sierra?)

### Versions
>

### Mention any other details that might be useful

> ---------------------------------------------------------------
> Thanks! We'll be in touch soon.

**Comment by Rickcau** [comment]
100% agree with this.   I tried setting this up locally and I get all sorts of errors when running the azd up.  One of the issues was with the docker part then once I got that fixed I am getting lots of other errors.

**Comment by Porkechebure** [comment]
The current source code is shit.
I don't use excuses and bland words, I say exactly what it is.
Many years into the field and since MS embraced OSS, all turned into shit.

They started to getting onto the bandwagon of all these shit js frameworks paradigms and now every application looks like a half baked inbreed between js frameworks and little of what the colossal and mature .NET framework was remained in it.

Now we are set to the js style:
Quintillions of config files and external tooling everywhere in every folder, totally untied from the code and you either have to get the ".NET fatigue" as it was for "Js fatigue" to read pages and pages of poorly written docs on the learn portal to be able to connect the wires togheter.

Same as of js, the whole introduction of a ton of tooling make it become like developing a react/angular app.
An ill, complex, fragile and immature ecosystem filled with immature tools written by anyone who think that can code.
You can resort to a bit of "standard" tooling that microsoft tries to set, but they make low effort to maintain them up to date because they (as every OSS) rely on community doing the work.

So now it's a total spaghetti code mess with nonsensical and unnecessary layers, injection forced everywhere with no real benefit just because "HEYYYY INJECTION IS COOOL WE ARE TOP EDGE DEVS IF WE USE IT"


No wonder you get any sort of errors running azd commands.
Heck, I also use projects that use azure, cosmos, openai services, blob storages, azure functions and everything else on azure, but I maintain a simple structure that had evolved little from what I was doing in the past, mostly to introduce only the REAL BENEFICIAL TOOLS, and not a ton of useless shit.
For example Docker that ease up lot of the environment setup or Linq which helps a ton (i know, it's old, it's just to mention) or minimal APIs that speed up the process of having an API.

My projects resemble much those of 10 years ago and everything works damn fine. A pity we can't use WCF anymore and we are forced with that BS rest, which is an excuse to mount up a lot of word salad over sending some json/xml strings over the http protocol and to make it the js way: untyped havoc with lot of errors, or for example all the HTTP verbs which are now "super duper standard if you don't use it you're a bad dev". They made a total hot dumster fire with them with all that FromBody, FromQuery and so on that barely connects well with all the http verb attributes (HttpPost, HttpGet etc) and causes troubles to everyone because params don't arrive in controllers.
And don't say it's me. Search on google and it's literally INFESTED with people looking up for help on those subjects.

**Comment by Coruscate5** [comment]
FWIW - you can deploy each resource to Azure individually, then starmap your resources back to IaC through bicep. We also had issues with the Docker architecture for Linux with private Nuget repos, so you can always just deploy your Website architecture to Windows App Service.

**Comment by Porkechebure** [comment]
> FWIW - you can deploy each resource to Azure individually, then starmap your resources back to IaC through bicep. We also had issues with the Docker architecture for Linux with private Nuget repos, so you can always just deploy your Website architecture to Windows App Service.

Lmao, they boast bicep, dev containers and all that sort of stuff to make the whole experience "seamless" than you have to do those shenaningans you are describing to deploy a simple app.

Really nice.

Oh and for the record, if you try to run "azd up" on a valid Ai Studio subscription it doesn't even work.

It gives

- --------------------------------------------------------------------------------
- RESPONSE 400: 400 Bad Request
- ERROR CODE: InvalidTemplate
- --------------------------------------------------------------------------------
- {
-   "error": {
-     "code": "InvalidTemplate",
-     "message": "Deployment template validation failed: 'The provided value for the template parameter 'openAiResourceGroupLocation' is not valid. The value 'centralus' is not part of the allowed value(s): 'canadaeast,eastus,eastus2,francecentral,switzerlandnorth,uksouth,japaneast,northcentralus,australiaeast'.'.",
-     "additionalInfo": [


A pity I didn't touch nothing and the Central US zone is what they put into it.
Lol.

**Comment by thomas-christiansen** [comment]
Totally agree. WHY do they have to over complicate things. Make a simple console app that illustrates the connection between the services, then developers can decide for them self to use docker, kubernetes, a space shuttle or what the h. they want.

**Comment by Timmate** [comment]
I can add that in my team Docker can't really be used as it's kind of blocked by the security team and getting access to it would take a while and wouldn't be that easy.
